molecule/default/tests/test_instance_per_network_per_hypervisor.py

```diff
@@ -57,59 +63,75 @@ def test_hypervisor_vms(host):
     image = filtered_images[-1]
 
     # neutron_agent connection lookup
-    na_list = testinfra.utils.ansible_runner.AnsibleRunner(
-        os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('neutron_agent')
-    hostname = host.check_output('hostname')
-    na_list = [x for x in na_list if hostname in x]
-    neutron_agent = next(iter(na_list), None)
-    assert neutron_agent
-    if 'container' in neutron_agent:
-        # lxc
-        na_pre = "lxc-attach -n {} -- bash -c ".format(neutron_agent)
-    else:
-        # ssh
-        na_pre = "ssh -o StrictHostKeyChecking=no \
-                   -o UserKnownHostsFile=/dev/null \
-                   {} ".format(neutron_agent)
+    # OSP use controller for nova_agent!
+    cmd = ('. /home/stack/stackrc && '
+           'openstack server show '
+           '-f value -c addresses '
+           'overcloud-controller-0')
+    res = host.run(cmd)
+    controller_network_name, controller_ip = res.stdout.split('=')
+    assert controller_ip
+    na_pre = ('ssh '
+              '-i /home/stack/.ssh/id_rsa '
+              '-o StrictHostKeyChecking=no '
+              '-o UserKnownHostsFile=/dev/null '
+              'heat-admin@{0} ').format(controller_ip)
+
+    # Copy rpc_support key to heat-admin account on overcloud-controller-0
+    cmd = ('scp '
+           '-i /home/stack/.ssh/id_rsa '
+           '-o StrictHostKeyChecking=no '
+           '-o UserKnownHostsFile=/dev/null '
+           '.ssh/rpc_support '
+           'heat-admin@{0}:.ssh/').format(controller_ip)
+    res = host.run(cmd)
+    assert res.rc == 0
```

Comment:
Lines 87 and 88 can be combined using run_expect as below:
```suggestion
    host.run_expect([0], cmd)
```

Reply:
Update pushed to adopt your suggestion @phongdly